### PR TITLE
Form/justin

### PR DESCRIPTION
**Pull request by juztinlazaro**
formserror

### CODE REVIEW

**Comment by srph** [comment]
IMO, it's much better if we only "redden" the border (minus the input bg). Where's the tooltip effect as you discussed?

**Comment by juztinlazaro** [comment]
erro message

Like this?

**Comment by srph** [comment]
I think we had a miscommunication. My bad. We already have a `-crimson` variant for our input. What we're missing is the tooltip-effect to show the errors.

pasted_image_at_2015_10_28_12_46_pm

Above, there's just an outline (`.-crimson`) when not focused. When focused, the tooltip would show.

**Comment by juztinlazaro** [comment]
dsa

**Comment by juztinlazaro** [comment]
dsa

File: styles/base/form.scss [hunk 1]
```diff
@@ -41,8 +41,54 @@
...
   }
+
+  &.-error {
+    @include form-variant($brand-crimson);
+    background-color: lighten($brand-crimson, 30);
+    color: $brand-crimson-dk;
+    &::-webkit-input-placeholder {
+      color: $brand-crimson-dk;
+    }
+  }
```
Comment (srph) [inline styles/base/form.scss]: Since we're not using this, can you remove this?

File: styles/base/form.scss [hunk 2]
```diff
@@ -41,8 +41,54 @@
...
+  height: 30px;
+  border-radius: 5px;
+}
+
+.form-group > .message:after {
+  content: '';
+  position: absolute;
+  bottom: 100%;
+  left: 50%;
+  margin-left: -8px;
```
Comment (srph) [inline styles/base/form.scss]: No need for this :+1:

File: styles/base/form.scss [hunk 3]
```diff
@@ -41,8 +41,54 @@
...
+}
+
+.form-group {
+  position: relative;
 }
 
+.form-group > .message {
+  position: absolute;
+  top: 40px;
+  left: 20px;
```
Comment (srph) [inline styles/base/form.scss]: No need for this :+1:

File: styles/base/form.scss [hunk 4]
```diff
@@ -41,8 +41,54 @@
...
+  width: 0;
+  height: 0;
+  border-bottom: 8px solid $brand-crimson;
+  border-right: 8px solid transparent;
+  border-left: 8px solid transparent;
+}
+
+.form-group .tooltip:focus + .message {
+  visibility: visible;
+}
```
Comment (srph) [inline styles/base/form.scss:79]: Just use the `form-control` selector, as in:

``` sass
.form-group .form-control + .message 
```

And remove the `.tooltip` selector :smile:

File: styles/base/form.scss [hunk 5]
```diff
@@ -41,8 +41,54 @@
...
+.form-group > .message {
+  position: absolute;
+  top: 40px;
+  left: 20px;
+  z-index: 9999;
+  visibility: hidden;
+  width: 220px;
+  text-align: center;
+  color: #FFFFFF;
+  line-height: 2;
```
Comment (srph) [inline styles/base/form.scss:58]: I think we'll have messages that will use multiple lines, so let's remove this rule.

File: styles/base/form.scss [hunk 6]
```diff
@@ -41,8 +41,54 @@
...
+  background: $brand-crimson;
+  height: 30px;
```
Comment (srph) [inline styles/base/form.scss]: Instead of height, why don't we use padding instead?

File: styles/base/form.scss [hunk 7]
```diff
@@ -41,8 +41,54 @@
...
+  position: relative;
 }
 
+.form-group > .message {
+  position: absolute;
+  top: 40px;
+  left: 20px;
+  z-index: 9999;
+  visibility: hidden;
+  width: 220px;
```
Comment (srph) [inline styles/base/form.scss]: `max-width` might be better :smile:

**Comment by srph** [comment]
Thanks, bae @juztinlazaro <3

**Pull request merged**